core/arch/x86/decode.c

```diff
@@ -1191,7 +1323,8 @@ decode_reg(decode_reg_t which_reg, decode_info_t *di, byte optype, opnd_size_t o
     switch (which_reg) {
     case DECODE_REG_REG:
         reg = di->reg;
-        extend = X64_MODE(di) && TEST(PREFIX_REX_R, di->prefixes);
+        extend = X64_MODE(di) && (TEST(PREFIX_REX_R, di->prefixes) ||
+                 TEST(PREFIX_EVEX_RR, di->prefixes)); /* for evex instructions */;
```

Comment:
Would need another, different 'extend' for EVEX R'? I think the meanings of R and R' overlap in 'extend' otherwise.